orchestrator/orchestratorAutomation.ps1

```diff
@@ -35,26 +39,50 @@ Param (
    [string] $storageport
 )
 
-Import-Module 'WebAdministration'
-    if(!$config){   
-        $config = "C:\Program Files (x86)\UiPath\Orchestrator\web.config"
-    }
+Import-Module 'WebAdministration' 
+    
 
-    if(!$aifport){   
-        $aifport = "31390"
-    }
+if(!$config){   
+    $config = "C:\Program Files (x86)\UiPath\Orchestrator"
+} 
 
-    if(!$storageport){
-        $storageport = "31443"
-    }
+#if path does not end with \ add it
+if( $config -notmatch '\\$' ){
+    $config += '\'
+}
 
-    if($domainBasedAccess.Length -gt 0){
-        $domainBasedAccess = $domainBasedAccess.ToString()
-    } else {
-        $domainBasedAccess = "false"
-    }
+$dll_config = $config + 'UiPath.Orchestrator.dll.config'
 
-echo "Path to web.config: "$config
+#Fetching Orchestrator version
+if(Test-Path $dll_config){
+    $orchestrator_version = [System.Diagnostics.FileVersionInfo]::GetVersionInfo($config + 'UiPath.Orchestrator.web.dll').FileVersion
+    echo "Orchestrator version : $orchestrator_version"
+}
+
+if(Test-Path $dll_config){

```

Comment:
Can we check if this file exists and if it has valid permissions ?

Reply:
that is what am checking with Test-Path i.e if the file exists and the script is supposed to be edited in powershell administrator mode so permission check is not required

Reply:
sure ... sounds good 👍